oap-server/server-storage-plugin/storage-iotdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/iotdb/IoTDBClient.java

```diff
@@ -0,0 +1,342 @@
+            }
+
+            List<String> columnNames = wrapper.getColumnNames();
+            IoTDBTableMetaInfo tableMetaInfo = IoTDBTableMetaInfo.get(modelName);
+            List<String> indexes = tableMetaInfo.getIndexes();
+            while (wrapper.hasNext()) {
+                Map<String, Object> map = new ConcurrentHashMap<>();
+                RowRecord rowRecord = wrapper.next();
+                List<Field> fields = rowRecord.getFields();
+                // transform timestamp to time_bucket
+                if (!UITemplate.INDEX_NAME.equals(modelName)) {
+                    map.put(IoTDBClient.TIME_BUCKET, TimeBucket.getTimeBucket(rowRecord.getTimestamp(),
+                            tableMetaInfo.getModel().getDownsampling()));
+                }
+                // field.get(0) -> Device, transform layerName to indexValue
+                String[] layerNames = fields.get(0).getStringValue().split("\\" + IoTDBClient.DOT + "\"");
+                for (int i = 0; i < indexes.size(); i++) {
+                    map.put(indexes.get(i), layerName2IndexValue(layerNames[i + 1]));
+                }
+                for (int i = 0; i < columnNames.size() - 2; i++) {
+                    String columnName = columnNames.get(i + 2);
+                    Field field = fields.get(i + 1);
+                    if (field.getDataType() == null) {
+                        continue;
+                    }
+                    if (field.getDataType().equals(TSDataType.TEXT)) {
+                        map.put(columnName, field.getStringValue());
+                    } else {
+                        map.put(columnName, field.getObjectValue(field.getDataType()));
+                    }
+                }
+                if (map.containsKey(IoTDBClient.NODE_TYPE_IDX)) {
+                    String nodeType = (String) map.get(IoTDBClient.NODE_TYPE_IDX);
+                    map.put(IoTDBClient.NODE_TYPE_IDX, Integer.valueOf(nodeType));
+                }
+                if (modelName.equals(BrowserErrorLogRecord.INDEX_NAME) || modelName.equals(LogRecord.INDEX_NAME)) {
+                    map.put(IoTDBClient.TIMESTAMP, map.get("\"" + IoTDBClient.TIMESTAMP + "\""));
+                }
+                for (Map.Entry<String, Object> entry : map.entrySet()) {
+                    // remove double quotes
+                    String key = entry.getKey();
+                    if (key.contains(".")) {
+                        map.put(key.substring(1, key.length() - 1), entry.getValue());
+                    }
+                }
+
+                storageDataList.add(storageBuilder.storage2Entity(map));
+            }
+            healthChecker.health();
+        } catch (IoTDBConnectionException | StatementExecutionException e) {
+            healthChecker.unHealth(e);
+            throw new IOException(e.getMessage() + System.lineSeparator() + "SQL Statement: " + querySQL, e);
+        } finally {
+            sessionPool.closeResultSet(wrapper);
```

Comment:
Add a null check for `wrapper` in case `wrapper = sessionPool.executeQueryStatement(querySQL);` throws exception and `wrapper` is still null, the same for many other places